core/src/main/scala/org/apache/gearpump/cluster/appmaster/AppMasterRuntimeEnvironment.scala

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.cluster.appmaster
+
+import akka.actor._
+import org.apache.gearpump.cluster.AppMasterToMaster.RegisterAppMaster
+import org.apache.gearpump.cluster.appmaster.AppMasterRuntimeEnvironment._
+import org.apache.gearpump.cluster.appmaster.ExecutorSystemScheduler.{Session, StartExecutorSystems}
+import org.apache.gearpump.cluster.appmaster.MasterConnectionKeeper.MasterConnectionStatus._
+import org.apache.gearpump.cluster.master.MasterProxy
+import org.apache.gearpump.cluster.{AppMasterContext, Application}
+import org.apache.gearpump.util.LogUtil
+
+/**
+ * This serves as runtime environment for AppMaster.
+ * When starting an AppMaster, we need to setup the connection to master, and prepare other environemnts.
+ *
+ * This also extend the function of Master, by providing a scheduler service for Executor System.
+ * AppMaster can ask Master for executor system directly. details like requesting resource,
+ * contacting worker to start a process, and then starting an executor system is hidden from AppMaster.
+ *
+ * Please use AppMasterRuntimeEnvironment.props() to construct this actor.
+ *
+ * @param appContextInput
+ * @param app
+ * @param masters
+ * @param masterFactory
+ * @param appMasterFactory
+ * @param masterConnectionKeeperFactory
+ */
+private[appmaster]
+class AppMasterRuntimeEnvironment (
+    appContextInput: AppMasterContext,
+    app: Application,
+    masters: Iterable[ActorPath],
+    masterFactory: (AppId, MasterActorRef) => Props,
+    appMasterFactory: (AppMasterContext, Application)=> Props,
+    masterConnectionKeeperFactory: (MasterActorRef, RegisterAppMaster, ListenerActorRef) => Props)
+  extends Actor {
```

Comment:
Maybe rename this class a better name because intuitively the "environment" should be static, not an actor.